lib/context.js

```diff
@@ -370,31 +379,42 @@ api.createTermDefinition = (activeCtx, localCtx, term, defined) => {
   }
 
   if('@container' in value) {
-    const container = value['@container'];
-    if(container !== '@list' && container !== '@set' &&
-      container !== '@index' && container !== '@language') {
-      let isValid = false;
-      const validContainers = ['@list', '@set', '@index', '@language'];
-      // JSON-LD 1.1 support
-      if(activeCtx['@version'] === 1.1) {
-        // || processingMode === 'jsonld-1.1') {
-        validContainers.push('@graph');
-        if(container === '@graph' ||
-          (_isArray(container) && container.length === 2 &&
-          container.includes('@graph') && container.includes('@set'))) {
-          isValid = true;
-        }
-      }
-      if(!isValid) {
-        throw new JsonLdError(
-          'Invalid JSON-LD syntax; @context @container value must be ' +
-          'one of the following: ' + validContainers.join(', '),
-          'jsonld.SyntaxError',
-          {code: 'invalid container mapping', context: localCtx});
-      }
+    // Normalize container to an array form
+    const container = _isString(value['@container']) ? [value['@container']] : (value['@container'] || []);
+    const validContainers = ['@list', '@set', '@index', '@language'];
+    let isValid = true;
+    let hasSet = container.includes('@set');
+
+    // JSON-LD 1.1 support
+    if(activeCtx.processingMode && activeCtx.processingMode.indexOf('json-ld-1.1') === 0) {
+      // TODO: @id and @type
+      validContainers.push('@graph');
+
+      // check container length
+      isValid &= container.length <= (hasSet ? 2 : 1);
+    } else {
+      // container must be a string which is one of the validContainers
```

Comment:
May want this comment to be a bit more explicit since the check doesn't directly involve strings:

```js
// container must not be an array (it must be a string, which is one of the validContainers)
```